save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/AgentsController.kt

```diff
@@ -41,10 +41,17 @@ class AgentsController(private val agentStatusRepository: AgentStatusRepository,
      */
     @PostMapping("/updateAgentStatusesWithDto")
     fun updateAgentStatusesWithDto(@RequestBody agentStates: List<AgentStatusDto>) {
-        agentStates.forEach {
-            val agent = agentRepository.findByContainerId(it.containerId)
-                ?: error("Agent with containerId=${it.containerId} not found in the DB")
-            agentStatusRepository.save(AgentStatus(it.time, it.state, agent))
+        agentStates.forEach { dto ->
+            val agent = agentRepository.findByContainerId(dto.containerId)
+                ?: error("Agent with containerId=${dto.containerId} not found in the DB")
+            val agentStatus = agentStatusRepository.findTopByAgentIdAndStateOrderByTimeDesc(agent.id!!, dto.state)
```

Comment:
Can we use `findTopByAgentContainerIdAndStateOrderByTimeDesc` instead and not make a query to get `agent`?